File: Python/credit.py
import cs50

#check input
num = cs50.get_int("Number: ")

#define variables
aux1, som, aux = 0, 0, 0

#convert int input to string, cut first 2 digits and convert back to int
n = str(num)
cod = int(n[ :2])

#calculates card digits according to Luhn's algorithm
while num > 0:
    aux1 += num % 10
    aux = ((num //10)%10)*2
    som += (aux//10 + aux%10)
    num = (num //10)//10
total = (aux1+som)%10

#check if it is valid and print result
if total == 0 and (cod == 34 or cod == 37):
    print("AMEX")
elif total == 0 and (cod >= 51 and cod <= 55):
    print("MASTERCARD")
elif total == 0 and cod//10 == 4:
    print("VISA")
else:
    print("INVALID")

File: C/credit.c
#include <stdio.h>
#include <cs50.h>

int main()
{   //define variables
    int digit1=0, digit2=0, sum=0, total, cod, i;
    //check input
    long int num = get_long ("Number: ");

    cod = num/100000000000000;
    //calculates card digits according to Luhn's algorithm
    while (num > 0){
       digit1 += num %10;
       digit2 = ((num /10)%10)*2;
       sum += (digit2/10 + digit2%10);
       num = (num /10)/10;
    }
    total = (digit1+sum)%10;
    //check if it is valid and print result
    if (total == 0 && (cod == 34 || cod == 37))
        printf("AMEX\n");
    else if (total == 0 && cod >= 51 && cod <= 55)
        printf("MASTERCARD\n");
    else if (total == 0 && cod/10 == 4)
        printf("VISA\n");
    else
        printf("INVALID\n");
}
